components/function-runtimes/java17/serverless-java-sdk/src/main/java/io/project/kyma/serverless/sdk/CloudEvent.java

```diff
@@ -0,0 +1,147 @@
+package io.project.kyma.serverless.sdk;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.logging.LoggingFeature;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.*;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CloudEvent {
+
+    private static final CloudEventHeaders[] CLOUD_EVENT_HEADERS = {CloudEventHeaders.CE_TYPE,
+            CloudEventHeaders.CE_SOURCE,
+            CloudEventHeaders.CE_EVENT_TYPE_VERSION,
+            CloudEventHeaders.CE_SPEC_VERSION,
+            CloudEventHeaders.CE_ID,
+            CloudEventHeaders.CE_TIME,};
+
+    private enum CloudEventHeaders {
+        CE_TYPE("ce-type"), CE_SOURCE("ce-source"), CE_EVENT_TYPE_VERSION("ce-eventtypeversion"),
+        CE_SPEC_VERSION("ce-specversion"), CE_TIME("ce-time"), CE_ID("ce-id");
+
+        private final String headerName;
+
+        public String getHeader() {
+            return this.headerName;
+        }
+
+        CloudEventHeaders(String name) {
+            this.headerName = name;
+        }
+    }
+
+    public ContainerRequestContext req;
+    private final MultivaluedMap<String, String> ceHeaders;
```

Comment:
Should we publish these headers? And what about other fields (maybe we keep something private but other runtimes published it)?

Reply:
I look at the Python runtime and we publish it (if we can say that something is hidden in python): https://github.com/kyma-project/kyma/blob/main/components/function-runtimes/python39/kubeless/ce.py#L44
I will make it public.